### PR TITLE
lava-docs: add controls section

### PR DESCRIPTION
**Pull request by seilagamo**
Add a section to describe controls and how to customize them.

### CODE REVIEW

File: .github/workflows/doc.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ on:
   workflow_dispatch:
   push:
     branches:
-      - main
+      - '*'
```
Comment (jroimartin) [inline .github/workflows/doc.yaml]: Can't you do the following?

```yaml
on: [workflow_dispatch, push]
```

File: src/README.md [hunk 2]
```diff
@@ -9,7 +9,8 @@ Lava is part of the [Vulcan ecosystem][vulcan-docs] and it is built on
 top of the same components that power Vulcan.
 Thus, Lava continuously benefits from Vulcan improvements.
 In fact, Lava is compatible with the [vulcan-checks][vulcan-checks]
-catalog shipped with Vulcan.
+catalog shipped with Vulcan. You can't find the list of security controls that
+Lava implements [here](controls.md).
```
Comment (jroimartin) [inline src/README.md]:
```suggestion
catalog shipped with Vulcan.

The default Lava configuration covers the following security controls:

- DAST (Dynamic Application Security Testing)
- SAST (Static Application Security Testing)
- SCA (Software Composition Analysis)
- Secret detection

For more details, please visit the [Security controls](controls.md)
section of the documentation.
```

File: build.bash [hunk 3]
```diff
@@ -20,11 +21,19 @@ add_main_header() {
 	sed -i "1i # ${header}" "${file}"
 }
 
+fetch_checktypes() {
+	local url="https://github.com/adevinta/lava-resources/releases/download/checktypes/${CHECKTYPES_VERSION}/checktypes.json"
+	curl -LsSf -o "build/checktypes.json" "${url}"
+}
```
Comment (jroimartin) [inline build.bash]: Why do you need this?

File: src/specs/checktype_catalog.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                            },
+                            "branch": {
+                                "type": "string"
+                            },
+                            "additionalProperties": true
+                        },
+                        "additionalProperties": false
+                    }
+                },
+                "required": ["name", "description", "image", "timeout", "assets"],
```
Comment (jroimartin) [inline src/specs/checktype_catalog.json]: Aren't the `description` and `timeout` fields optional?

File: src/controls.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
+# Controls and customization
+
+Behind the scenes Lava makes use of other security tools and custom security
+checks to perform a security scan. Tools are integrated through what is called
+["checktypes"](helpdoc/checktypes.md).
+
+Checktypes are part of the Vulcan ecosystem and both Vulcan and Lava share the
+same collection located at <https://github.com/adevinta/vulcan-checks>
```
Comment (jroimartin) [inline src/controls.md]:
```suggestion
same collection located at <https://github.com/adevinta/vulcan-checks>.
```

File: src/controls.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
+# Controls and customization
+
+Behind the scenes Lava makes use of other security tools and custom security
```
Comment (jroimartin) [inline src/controls.md]:
```suggestion
Behind the scenes Lava uses other security tools and custom security
```

File: src/controls.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+checks to perform a security scan. Tools are integrated through what is called
+["checktypes"](helpdoc/checktypes.md).
+
+Checktypes are part of the Vulcan ecosystem and both Vulcan and Lava share the
+same collection located at <https://github.com/adevinta/vulcan-checks>
+
+## Type of controls
```
Comment (jroimartin) [inline src/controls.md]:
```suggestion
## Types of security controls
```

File: src/controls.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+| SCA                                    | Software Composition Analysis. For instance, detection of outdated dependencies.                                                                                    |
+| Secret detection                       |                                                                                                                                                                     |
+| Finding collection                     | This category referees to checktypes that do not run the analysis themselves. Instead, they collect the findings from 3rd party systems like Tenable.io or vulners. |
+| Recon                                  | Checktypes focused on reconnaissance activities. For instance, port scanning and service discovery.                                                                 |
+| Data analysis                          | Checktypes focused on processing already available data to produce results from the point of view of information security.                                          |
+
```
Comment (jroimartin) [inline src/controls.md:25]: Remove extra newline.

```suggestion
```

File: src/controls.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+checks to perform a security scan. Tools are integrated through what is called
+["checktypes"](helpdoc/checktypes.md).
+
+Checktypes are part of the Vulcan ecosystem and both Vulcan and Lava share the
+same collection located at <https://github.com/adevinta/vulcan-checks>
+
+## Type of controls
+
+This table classifies each checktype based on the security controls that it
+covers.
```
Comment (jroimartin) [inline src/controls.md]:
```suggestion
The following table lists all the security controls covered by Lava's
default checktype collection.
```

File: src/controls.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+| <div style="width:175px">Control</did> | Description                                                                                                                                                         |
```
Comment (jroimartin) [inline src/controls.md]: I would use mdbook's default style and rename the column to `Security Control`.

```suggestion
| Security Control | Description                                                                                                                                                         |
```

File: src/controls.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+| SAST - Application                     | Static analysis of application-related assets. Inother words, security analysis of the source code of an application.                                               |
+| SAST - IaC                             | Static analysis of Infrastructure as Code manifests. For instance, AWS CloudFormation templates or Terraform configurations.                                        |
+| SCA                                    | Software Composition Analysis. For instance, detection of outdated dependencies.                                                                                    |
+| Secret detection                       |                                                                                                                                                                     |
+| Finding collection                     | This category referees to checktypes that do not run the analysis themselves. Instead, they collect the findings from 3rd party systems like Tenable.io or vulners. |
```
Comment (jroimartin) [inline src/controls.md]:
```suggestion
| Finding collection                     | This category refers to checktypes that do not run the analysis themselves. Instead, they collect the findings from 3rd party systems like Tenable.io or Vulners. |
```

File: src/controls.md [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+| Recon                                  | Checktypes focused on reconnaissance activities. For instance, port scanning and service discovery.                                                                 |
+| Data analysis                          | Checktypes focused on processing already available data to produce results from the point of view of information security.                                          |
```
Comment (jroimartin) [inline src/controls.md]:
```suggestion
| Data analysis                          | Checktypes focused on processing data sets to extract information considered relevant from an information security point of view.                                          |
```

File: src/controls.md [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+  </tbody>
+</table>
+
+Since the check versions are updated regularly in an automatic way, the best way
+to know the version that Lava is using is checking the source code of each
+check.
+
+## Disabled by Default
+
+| Checktype                                                                                                            | Description                                                                             | Type of Control                     |
```
Comment (jroimartin) [inline src/controls.md]:
```suggestion
| Checktype                                                                                                            | Description                                                                             | Security Control                     |
```

File: src/controls.md [hunk 14]
```diff
@@ -0,0 +1,139 @@
+# Controls and customization
```
Comment (jroimartin) [inline src/controls.md]:
```suggestion
# Security Controls
```

File: src/controls.md [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+| SCA                                    | Software Composition Analysis. For instance, detection of outdated dependencies.                                                                                    |
+| Secret detection                       |                                                                                                                                                                     |
+| Finding collection                     | This category referees to checktypes that do not run the analysis themselves. Instead, they collect the findings from 3rd party systems like Tenable.io or vulners. |
+| Recon                                  | Checktypes focused on reconnaissance activities. For instance, port scanning and service discovery.                                                                 |
+| Data analysis                          | Checktypes focused on processing already available data to produce results from the point of view of information security.                                          |
+
+
+## Enabled by Default
```
Comment (jroimartin) [inline src/controls.md]:
```suggestion
## Enabled by default
```

File: src/controls.md [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+      <td>DAST - Application</td>
+    </tr>
+  </tbody>
+</table>
+
+Since the check versions are updated regularly in an automatic way, the best way
+to know the version that Lava is using is checking the source code of each
+check.
+
+## Disabled by Default
```
Comment (jroimartin) [inline src/controls.md]:
```suggestion
## Disabled by default
```

File: src/controls.md [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+
+
+## Enabled by Default
+
+<table>
+  <thead>
+    <tr>
+      <th>Checktype</th>
+      <th>Description</th>
+      <th>Type of Control</th>
```
Comment (jroimartin) [inline src/controls.md]:
```suggestion
      <th>Security Control</th>
```

File: src/controls.md [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+    <tr>
+      <td>
+        <a href="https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-nuclei">vulcan-nuclei</a>
+      </td>
+      <td>
+        Scan web addresses with projectdiscovery/nuclei.<br>
+        Official Documentation:
+        <a href="https://docs.projectdiscovery.io/tools/nuclei"
+          >https://docs.projectdiscovery.io/tools/nuclei</a
+        >
```
Comment (jroimartin) [inline src/controls.md]: I would use markdown tables and single-line descriptions. E.g.

```
Scan web addresses with projectdiscovery/nuclei. Official Documentation: <https://docs.projectdiscovery.io/tools/nuclei>.
```

**Review by jroimartin — CHANGES_REQUESTED** [review]

**Comment by janelastur** [comment]
Observations about the section of security controls.

- We should provide an order and sense of importance with the controls for the user lecture. An order of importance can be SAST --> SCA --> IAC --> Secret Detection --> DAST - Application --> rest of the controls not enabled by default. This should be the order because these are the ones enabled by default. So the first table "Types of controls" should be this list top to bottom. In the second table "Enabled by default", it should be from top to bottom SAST --> SCA --> IAC --> Secret Detection --> DAST - Application.
- "IaC" is its own category of control. It should be called "IaC" and not "SAST - IaC". This implies that there is only one type of SAST and it should be called "SAST" (and not "SAST - Application").
- In the "Type of controls - Description" Column we should standardise the format. Some of the descriptions is just the acronym, some are the acronym + an additional explanation and some are empty. In my opinion, all should have acronym + additional explanation.

**Comment by seilagamo** [comment]
I do not agree with changing "SAST - IaC" with IaC because IaC (Infrastructure as Code) doesn't explain what kind of analysis the tool is performing. The definition of IaC is "the ability to provision and support your computing infrastructure using code instead of manual processes and settings" and is not a security control itself.

File: src/controls.md [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+| [vulcan-trivy](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-trivy)       | Scan docker images and Git repositories using aquasec/trivy. Official Documentation: <https://aquasecurity.github.io/trivy/latest> | SCA, Secret Detection, SAST - IaC |
+| [vulcan-retirejs](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-retirejs) | Check web pages for vulnerable JavaScript libraries. Official Documentation: <https://retirejs.github.io/retire.js>                | SCA                               |
+| [vulcan-nuclei](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-nuclei)     | Scan web addresses with projectdiscovery/nuclei. Official Documentation: <https://docs.projectdiscovery.io/tools/nuclei>           | DAST - Application                |
+| [vulcan-zap](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-zap)           | Runs an OWASP ZAP passive or active scan. Official Documentation: <https://www.zaproxy.org/docs>                                   | DAST - Application                |                                                                                 |
```
Comment (jroimartin) [inline src/controls.md]:
```suggestion
| [vulcan-zap](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-zap)           | Runs an OWASP ZAP passive or active scan. Official Documentation: <https://www.zaproxy.org/docs>                                   | DAST - Application                |
```

**Review by jroimartin — CHANGES_REQUESTED** [review]

File: src/controls.md [hunk 20]
```diff
@@ -10,65 +10,62 @@ same collection located at <https://github.com/adevinta/vulcan-checks>.
...
-| Data analysis         | Checktypes focused on processing data sets to extract information considered relevant from an information security point of view.                                 |
+| Security Control   | Description                                                                                                                                                       |
+|--------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| SAST               | Static Application Security Testing. It's the analysis of source code.                                                                                            |
+| SCA                | Software Composition Analysis. For instance, detection of outdated dependencies.                                                                                  |
+| IaC Security       | Static analysis of Infrastructure as Code manifests. For instance, AWS CloudFormation templates or Terraform configurations.                                      |
```
Comment (jroimartin) [inline src/controls.md]: For consistency with the other security controls.

```suggestion
| IaC security       | Static analysis of Infrastructure as Code manifests. For instance, AWS CloudFormation templates or Terraform configurations.                                      |
```

File: src/controls.md [hunk 21]
```diff
@@ -10,65 +10,62 @@ same collection located at <https://github.com/adevinta/vulcan-checks>.
...
+| Checktype                                                                                    | Description                                                                                                                        | Security Control                    |
+|----------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------|
+| [vulcan-semgrep](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-semgrep)   | Finds potential issues in source code from a Git repository using Semgrep. Official Documentation: <https://semgrep.dev/docs>      | SAST                                |
+| [vulcan-trivy](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-trivy)       | Scan docker images and Git repositories using aquasec/trivy. Official Documentation: <https://aquasecurity.github.io/trivy/latest> | SCA, Secret Detection, IaC Security |
```
Comment (jroimartin) [inline src/controls.md]: For consistency with the previous table.

```suggestion
| [vulcan-trivy](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-trivy)       | Scan docker images and Git repositories using aquasec/trivy. Official Documentation: <https://aquasecurity.github.io/trivy/latest> | SCA, IaC security, Secret detection |
```

File: src/controls.md [hunk 22]
```diff
@@ -10,65 +10,62 @@ same collection located at <https://github.com/adevinta/vulcan-checks>.
...
-| [vulcan-tenable](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-tenable)                           | Retrieves findings from assets scanned by Tenable.io.                                   | Finding Collection                  |
-| [vulcan-vulners](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-vulners)                           | Runs the nmap-script vulners against the target opened ports.                           | DAST - Infrastructure               |
-| [vulcan-wpscan](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-wpscan)                             | Runs Wordpress scan.                                                                    | DAST - Application                  |
+| Checktype                                                                                                            | Description                                                                             | Security Control   |
```
Comment (jroimartin) [inline src/controls.md:42]: Use the same case as in the first table (i.e. `Data analysis`, `Finding collection`, `Secret detection`).

**Review by jroimartin — CHANGES_REQUESTED** [review]
I added a couple of minor comments about the style. Please, fix them and clean-up the commit history. Also, delete the temporary changes in the workflow and the `build.bash` script.

**Review by jroimartin — APPROVED** [review]
LGTM

**Pull request merged**